pkg/controller/reconciler.go

```diff
@@ -133,16 +136,30 @@ func (r *reconciler) delete(ctx context.Context, obj client.Object, logger logr.
 	}
 
 	logger.Info("Reconciling object deletion")
-	if err := r.actuator.Delete(ctx, obj); err != nil {
+	requeueAfter, err := r.actuator.Delete(ctx, obj)
+	if err != nil {
 		return reconcileErr(errors.Wrap(err, "could not reconcile object deletion"))
 	}
 	logger.Info("Successfully reconciled object deletion")
 
 	logger.Info("Removing finalizer")
-	if err := controllerutils.RemoveFinalizer(ctx, r.reader, r.client, obj, r.finalizerName); err != nil {
+	if err := controllerutils.RemoveFinalizer(ctx, r.reader, r.client, obj, r.finalizerName); client.IgnoreNotFound(err) != nil {
 		return reconcile.Result{}, errors.Wrap(err, "could not remove finalizer")
 	}
 
+	if requeueAfter != time.Duration(0) {
+		return reconcile.Result{RequeueAfter: requeueAfter}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) ensureDeleted(ctx context.Context, obj client.Object, logger logr.Logger) (reconcile.Result, error) {
+	logger.Info("Ensuring object deletion")
+	if _, err := r.actuator.Delete(ctx, obj); err != nil {
```

Comment:
I guess the `err` resulted from `Delete()` will nil in case that the object does not exists?